scylla-cql/src/deserialize/value.rs

```diff
@@ -902,6 +935,220 @@ where
     }
 }
 
+pub struct ConstLengthVectorIterator<'frame, 'metadata, T> {
+    coll_typ: &'metadata ColumnType<'metadata>,
+    elem_typ: &'metadata ColumnType<'metadata>,
+    remaining: usize,
+    elem_len: usize,
+    slice: FrameSlice<'frame>,
+    phantom_data: std::marker::PhantomData<T>,
+}
```

Comment:
You implemented 2 vector iterator types: one for vectors with know length and one for those without.
User needs to use the correct one depending on the type of vector element. 

As far as I can tell there is no difference in user-facing API between the two, they just do deserialization differently internally.

Can we have just one type, which will handle different vector types internally?